tests/libvmi/vmi.go

```diff
@@ -20,16 +20,25 @@
 package libvmi
 
 import (
+	"fmt"
+
 	k8sv1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/resource"
 	k8smetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/rand"
-
 	v1 "kubevirt.io/api/core/v1"
 
+	"kubevirt.io/kubevirt/tests/flags"
+
 	"kubevirt.io/kubevirt/pkg/pointer"
 )
 
+const (
+	hookSidecarImage                   = "example-hook-sidecar"
+	annotationKeyHookSideCars          = "hooks.kubevirt.io/hookSidecars"
```

Comment:
Depending on the abstraction level, these should either not exist in libvmi or to be placed in a file that is focused on sidecars hooks.

I see people polluted this file with time, adding various things in it without considering where it is better fit.

Reply:
If that is the abstraction layer you are looking for, then yes.
The builders will have to fit that abstraction, e.g. `WithSidecarHook(…)`. It should add the input correctly to the right locations in the vmi object. It needs to be generic to any future sidecar hook.